pkg/planner/core/rule_decorrelate.go

```diff
@@ -204,12 +204,16 @@ func (s *DecorrelateSolver) Optimize(ctx context.Context, p base.LogicalPlan, op
 		innerPlan := apply.Children()[1]
 		apply.CorCols = coreusage.ExtractCorColumnsBySchema4LogicalPlan(apply.Children()[1], apply.Children()[0].Schema())
 		if len(apply.CorCols) == 0 {
-			// If the inner plan is non-correlated, the apply will be simplified to join.
-			join := &apply.LogicalJoin
-			join.SetSelf(join)
-			join.SetTP(plancodec.TypeJoin)
-			p = join
-			appendApplySimplifiedTraceStep(apply, join, opt)
+			if p.SCtx().GetSessionVars().EnableCascadesPlanner {
+				// left the transformation to cascades.
```

Comment:
this is just a temp solution to keep original apply to test the xformed op should derive their stats own. 

by this way, we can keep the original apply and xformed join at same time, while the original apply couldn't generate the index range scan any more because the correlated condition is pulled into the apply itself rather than a selection just above the ds. ref the picture below which will be solved in the next pull request

![image](https://github.com/user-attachments/assets/ea8f1907-ac63-462d-8f29-6aef7eac2069)